src/crud.py

```
@@ -0,0 +1,110 @@
+    if page > 1:
+        prev_page = f"/theater/movies/?page={page - 1}&per_page={per_page}"
+
+    next_page = None
+    if page < total_pages:
+        next_page = f"/theater/movies/?page={page + 1}&per_page={per_page}"
+
+    return MovieListSchema(
+        movies=movies,
+        prev_page=prev_page,
+        next_page=next_page,
+        total_pages=total_pages,
+        total_items=total_items,
+    )
+
+
+def create_movie(movie_data: MovieCreateBaseSchema, db: Session) -> MovieModel:
+    country = db.query(CountryModel).filter(CountryModel.code == movie_data.country).first()
+
+    if not country:
+        country = CountryModel(code=movie_data.country)
+        db.add(country)
+        db.flush()
+
+    genres = [
+        db.query(GenreModel).filter(GenreModel.name == name).first() or GenreModel(name=name)
+        for name in movie_data.genres
+    ]
+    actors = [
+        db.query(ActorModel).filter(ActorModel.name == name).first() or ActorModel(name=name)
+        for name in movie_data.actors
+    ]
+    languages = [
+        db.query(LanguageModel).filter(LanguageModel.name == name).first() or LanguageModel(name=name)
+        for name in movie_data.languages
+    ]
+
+    new_movie = MovieModel(
+        **movie_data.model_dump(exclude={"country", "genres", "actors", "languages"}),
+        country=country,
+        genres=genres,
+        actors=actors,
+        languages=languages,
+    )
+
+    db.add(new_movie)
+    db.commit()
+    db.refresh(new_movie)
+
+    return new_movie
+
+
+def get_movie_by_id(movie_id: int, db: Session) -> MovieModel | None:
+    return db.query(MovieModel).filter(MovieModel.id == movie_id).first()
+
+
+def get_movie_by_name_and_date(name: str, date: date, db: Session) -> MovieModel | None:
+    return db.query(MovieModel).filter(MovieModel.name == name, MovieModel.date == date).first()
+
+
+def update_movie(
+        movie: MovieModel,
+        movie_data: MovieUpdateSchema,
+        db: Session
+) -> MovieModel:
```

The `update_movie` function should include error handling to check if the movie exists before attempting to update it. If the movie does not exist, it should raise an `HTTPException` with a `404 Not Found` status code and an appropriate message.